src/sql/src/plan/statement/ddl.rs

```diff
@@ -2725,14 +2773,33 @@ impl PostgresConnectionOptionExtracted {
             None
         };
 
+        let host = match (self.host, self.aws_privatelink) {
+            (None, None) => {
+                sql_bail!("invalid connection: must specify either HOST or AWS PRIVATELINK option");
+            }
+            (Some(_), Some(_)) => {
+                sql_bail!(
+                    "invalid connection: must specify only one of HOST or AWS PRIVATELINK options"
+                );
+            }
+            (Some(host), None) => host,
+            (None, Some(aws_privatelink)) => {
+                let id = GlobalId::from(aws_privatelink);
+                let entry = scx.catalog.get_item(&id);
+                match entry.connection()? {
+                    // TODO: don't hardcode connection hostname logic.
+                    Connection::AwsPrivatelink(_) => format!("connection-{id}"),
+                    _ => sql_bail!("{} is not an AWS PRIVATELINK connection", entry.name().item),
+                }
+            }
+        };
+
         Ok(mz_storage::types::connections::PostgresConnection {
             database: self
```

Comment:
Database is now only required if `host` is provided, so that check should be moved into the `self.host` match block

Reply:
(Refactored away.)